autotst/reaction.py

```diff
@@ -359,58 +361,87 @@ def get_labeled_reaction(self):
                     rmg_products.append(prod.molecule)
                 elif isinstance(prod, RMGMolecule):
                     rmg_products.append([prod])
-
             test_reactants = []
             test_products = []
             if len(rmg_reactants) == 1:
-                test_reactants = rmg_reactants
+                for l1 in rmg_reactants[0]:
+                    test_reactants.append([l1])
             elif len(rmg_reactants) == 2:
                 l1, l2 = rmg_reactants
                 for i in l1:
                     for j in l2:
                         test_reactants.append([i, j])
 
             if len(rmg_products) == 1:
-                test_reactants = rmg_products
+                for l1 in rmg_products[0]:
+                    test_products.append([l1])
             elif len(rmg_products) == 2:
                 l1, l2 = rmg_products
                 for i in l1:
                     for j in l2:
                         test_products.append([i, j])
+
+            reacts = test_reactants[:]
+            prods = test_products[:]
             for name, family in list(self.rmg_database.kinetics.families.items()):
+                logging.info("Trying to match {} to {}".format(self.rmg_reaction, family))
+
                 if match:
-                    break
+                    continue
+                
+                test_reactants = reacts[:]
+                test_products = prods[:]
                 for test_reactant in test_reactants:
-                    for test_products in test_products:
+                    for test_product in test_products:
 
                         if match:
                             continue
 
                         test_reaction = RMGReaction(
-                            reactants=test_reactant, products=test_products)
+                            reactants=test_reactant, products=test_product)
+
+                        try:
+                            labeled_r, labeled_p = family.getLabeledReactantsAndProducts(
+                                test_reaction.reactants, test_reaction.products)
+                            if not (labeled_r and labeled_p):
+                                raise ActionError
+                        except (ValueError, ActionError, IndexError):
+                            try: 
+                                # Trying the reverse reaction if the forward reaction doesn't work
```

Comment:
maybe add logging info to let the user know that the forward direction failed, so we are trying the backwards direction.

Reply:
I added the logging requested